walk/config.go

```diff
@@ -73,18 +74,48 @@ func (wc *walkConfig) shouldFollow(p string) bool {
 
 var _ config.Configurer = (*Configurer)(nil)
 
-type Configurer struct{}
+type Configurer struct {
+	// Excludes and BUILD filenames specified on the command line.
+	// May be extending with BUILD directives.
+	cliExcludes       []string
+	cliBuildFileNames string
 
-func (*Configurer) RegisterFlags(fs *flag.FlagSet, cmd string, c *config.Config) {
-	wc := &walkConfig{}
-	c.Exts[walkName] = wc
-	fs.Var(&gzflag.MultiFlag{Values: &wc.excludes}, "exclude", "pattern that should be ignored (may be repeated)")
+	// Alternate BUILD read/write directories
+	readBuildFilesDir, writeBuildFilesDir string
 }
 
-func (*Configurer) CheckFlags(fs *flag.FlagSet, c *config.Config) error { return nil }
+func (wc *Configurer) RegisterFlags(fs *flag.FlagSet, cmd string, c *config.Config) {
+	fs.Var(&gzflag.MultiFlag{Values: &wc.cliExcludes}, "exclude", "pattern that should be ignored (may be repeated)")
+	fs.StringVar(&wc.cliBuildFileNames, "build_file_name", strings.Join(config.DefaultValidBuildFileNames, ","), "comma-separated list of valid build file names.\nThe first element of the list is the name of output build files to generate.")
+	fs.StringVar(&wc.readBuildFilesDir, "experimental_read_build_files_dir", "", "path to a directory where build files should be read from (instead of -repo_root)")
+	fs.StringVar(&wc.writeBuildFilesDir, "experimental_write_build_files_dir", "", "path to a directory where build files should be written to (instead of -repo_root)")
+}
+
+func (wc *Configurer) CheckFlags(_ *flag.FlagSet, c *config.Config) error {
+	c.ValidBuildFileNames = strings.Split(wc.cliBuildFileNames, ",")
+	if wc.readBuildFilesDir != "" {
+		if filepath.IsAbs(wc.readBuildFilesDir) {
+			c.ReadBuildFilesDir = wc.readBuildFilesDir
+		} else {
+			c.ReadBuildFilesDir = filepath.Join(c.WorkDir, wc.readBuildFilesDir)
+		}
+	}
+	if wc.writeBuildFilesDir != "" {
+		if filepath.IsAbs(wc.writeBuildFilesDir) {
+			c.WriteBuildFilesDir = wc.writeBuildFilesDir
+		} else {
+			c.WriteBuildFilesDir = filepath.Join(c.WorkDir, wc.writeBuildFilesDir)
+		}
+	}
+
+	c.Exts[walkName] = &walkConfig{
```

Comment:
Note this assignment was moved from `RegisterFlags` to `CheckFlags`. Mainly just to make all these cli params more consistent, but imo this makes it clearer what comes from the CLI args (all put into `Configurer` fields) vs later getting appended from directives.

Reply:
It probably doesn't matter since they're nearly called at the same time, but the doc for `RegisterFlags` indicates it's supposed to add to `c.Exts`.

Is there a reason to track excludes separately if they're set by flags or directives? Previously they were appended to the same slice.

Reply:
I mainly did it just to align with how `config/config` was doing it. Now both here and in `config/config` the cli data is stored in fields and put into `c.*` and `c.Exts` at the same time 🤷 

I can change it if you have a preference.